server/chat/backend/agent/utils/cache_control.py

```
@@ -0,0 +1,107 @@
+    return ttl if ttl in _VALID_TTLS else "5m"
+
+
+def _is_anthropic_model(model_name: str) -> bool:
+    if not model_name:
+        return False
+    # OpenAI/Gemini auto-cache server-side and ignore (or reject) cache_control,
+    # so we only tag Anthropic-family models. Match the OpenRouter id prefix.
+    return model_name.split("/", 1)[0].lower() == "anthropic"
+
+
+def build_cached_system_prompt(model_name: str, system_prompt_text: str) -> Any:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `build_cached_system_prompt`

(ANN401)